haas/ext/obm/ipmi.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2013-2014 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+A obm driver for implementing lego robot based controls for 
+arm based servers on in a DC at North Pole :-)
+
+"""
+
+from sqlalchemy import Column, String, Integer, ForeignKey
+import schema
+
+
+from haas.model import Obm
+from haas.errors import OBMError
+from haas.dev_support import no_dry_run
+
+class Ipmi(Obm):
+    id = Column(Integer, ForeignKey('obm.id'), primary_key=True)
+    ipmi_host = Column(String, nullable=False) 
+    ipmi_user = Column(String, nullable=False)
+    ipmi_password = Column(String, nullable=False)
+
+    api_name = 'http://schema.massopencloud.org/haas/v0/obm/ipmi'
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+        }
+    @staticmethod
+    def validate(kwargs):
+        schema.Schema({
+            'type': Ipmi.api_name,
+            'ipmi_host': basestring,
+            'ipmi_user': basestring,
+            'ipmi_password': basestring,
+            }).validate(kwargs)
+
+    def _ipmitool(self, args):
+        """ Invoke ipmitool with the right host/pass etc. for this code.
```

Comment:
s/code/node/

Also, scan over [PEP 257][1] and adjust the formatting of your docstrings accordingly; we follow the conventions described there.

[1]: https://www.python.org/dev/peps/pep-0257/